hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/Options.java

```diff
@@ -518,4 +522,94 @@ public String toString() {
     MD5MD5CRC,  // MD5 of block checksums, which are MD5 over chunk CRCs
     COMPOSITE_CRC  // Block/chunk-independent composite CRC
   }
+
+  /**
+   * The standard {@code openFile()} options.
+   */
+  @InterfaceAudience.Public
+  @InterfaceStability.Evolving
+  public static final class OpenFileOptions {
+
+    private OpenFileOptions() {
+    }
+
+    /**
+     * Prefix for all standard filesystem options: {@value}.
+     */
+    public static final String FILESYSTEM_OPTION = "fs.option.";
+
+    /**
+     * Prefix for all openFile options: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE =
+        FILESYSTEM_OPTION + "openfile.";
+
+    /**
+     * OpenFile option for file length: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_LENGTH =
+        FS_OPTION_OPENFILE + "length";
+
+    /**
+     * OpenFile option for split start: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_SPLIT_START =
+        FS_OPTION_OPENFILE + "split.start";
+
+    /**
+     * OpenFile option for split end: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_SPLIT_END =
+        FS_OPTION_OPENFILE + "split.end";
+
+    /**
+     * OpenFile option for buffer size: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_BUFFER_SIZE =
+        FS_OPTION_OPENFILE + "buffer.size";
+
+    /**
+     * OpenFile option for seek policies: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_FADVISE =
+        FS_OPTION_OPENFILE + "fadvise";
+
+    /**
+     * fadvise policy 'normal' -up to implementation: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_FADVISE_NORMAL =
+        "normal";
+
+    /**
+     * fadvise policy for sequential IO: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_FADVISE_SEQUENTIAL =
+        "sequential";
+
+    /**
+     * fadvise policy for random: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_FADVISE_RANDOM =
+        "random";
+
+    /**
+     * fadvise policy for adaptive IO: {@value}.
+     */
+    public static final String FS_OPTION_OPENFILE_FADVISE_ADAPTIVE =
+        "adaptive";
+
+    /**
+     * Set of standard options which openFile implementations
+     * MUST recognize, even if they ignore the actual values.
```

Comment:
See my comment in the builder about possibly making these properties (with getter and setter).